### DISCUSSION TITLE
Validate if session is valid when using authstore.save

### DISCUSSION THREAD

**Discussion by LickABrick** [pocketbase/pocketbase#2408] [Q&A] [answered]
Maybe I'm doing things wrong, but I have a NestJS with BullMQ app running in my backend. This handles background/scheduled tasks for the user.

I want the tasks to run on behalf of the user that invoked the task. When submitting a POST request to my NestJS server from my publicly available app I also send the JWT token as a header option. In NestJS I want to validate this JWT token.

_(The NestJS server is not publicly available, requests are send serverside from my SvelteKit app, so over LAN)_

The weird thing is (or maybe there is a valid reason for this) that authStore.isValid returns true if I insert invalid JWT. Examples:

```
const JWT = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c' // Valid JWT token
pb.authStore.save(JWT, null)
console.log(pb.authStore.isValid) // true
```
```
const JWT = 'invalideyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c' // invalid JWT token, notice the start
pb.authStore.save(JWT, null)
console.log(pb.authStore.isValid) // true
```
```
const JWT = '1234' // invalid JWT token
pb.authStore.save(JWT, null)
console.log(pb.authStore.isValid) // false
```
Is this expected behaviour? If so how can I properly validate if the JWT token belongs to an existing user in my PB instance?
If this is not possible would a 'workaround' be that in my validation function in NestJS I query a dummy record from PocketBase which requires a user to be authenticated, if the record is returned the token is valid and the user may use my NestJS instance? 

Let me know if anything is unclear.

**Comment by ganigeorgiev** [comment, marked as answer]
`pb.authStore` represents only a local state, aka. it doesn't perform http requests.

This means that `pb.authStore.isValid` only _loosely_ checks if the store has nonempty jwt token with  valid payload (eg. whether it has an `exp` claim and it is not expired).

The actual token validation is performed when you send request to PocketBase.

If you want to check whether the loaded `pb.authStore.token` is valid as part of your NestJS app, you can load it into the store and then send a `pb.collection('users').authRefresh()` call. On success this will automatically refresh the store with the latest user data and a new token with renewed exp claim. On missing, invalid or expired token - throws an error.

**Reply by LickABrick** [reply]
Thanks, that is exactly what I was looking for!!

**Reply by twiddlecoding** [reply]
How do you load the token into the store?

**Reply by ganigeorgiev** [reply]
@twiddlecoding by calling `pb.authStore.save(token)`. For all store methods please refer to https://github.com/pocketbase/js-sdk#common-auth-store-fields-and-methods.
